Input/include/ParseParams.hpp

```diff
@@ -39,6 +39,9 @@ class ParseParams
 	 */
 	ParamsCarrier *getParams();
 
+	void newSettingsLocationCheck(ParamsCarrier *singleTone, std::string settings);
```

Comment:
I think separation of these three functions is OK. However, I would suggest to move them into separate class / file. 

The most suitable place would be Utils package - create a new validator there for input. This would allow us to validate input values anywhere we want.